### ISSUE TITLE
GetChannelMessagesAsync not working as intended?

### ISSUE THREAD

**Issue by Xeevis** [RogueException/Discord.Net#115]
Intuitively in order to get last 1000 messages from a channel you'd write and I believe that's intended functionality

    await msg.Channel.GetMessagesAsync(1000);
This is however not working correctly, it will get roughly last 100 messages and they'll just repeat like 10 times.

*dev*

**Comment by khionu** [comment]
That's because there is a limit of 100 messages :P The bad handling.... might be another issue

**Comment by RogueException** [comment]
I was able to reproduce, looking into it.

**Comment by Xeevis** [comment]
@khionu That is limit per request, with proper use of direction and relative message id for pagination, it should be possible to get last 1000 messages with 10 sequential requests. And it does make 10 requests, problem is that it returns 10 very similar pages. :)

**Comment by RogueException** [comment]
I have a fix for this ready but stumbled across another bug that needs clarification from the discord devs, I'll get back to you later on this.

**Comment by Xeevis** [comment]
@RogueException What kind of bug? Can you please attach me your fix, I'm curious :smile:

**Comment by 420foxbot** [comment]
Preliminary fix should be in ee04e1ddcce571d1351322681bd0cc083b37f8f0

**Comment by RogueException** [comment]
Last part of the fix is pushed to dev.

The issue I was having was that I have a rare case of Discord not giving me messages in order. It appears that if two messages are sent in an extremely short time (they actually appear to have the same timestamp), the ordering may break, which can cause duplicate messages to be returned.

I added a quick check that scans the results to make sure it has the highest or lowest id before doing doing the next request, if count is >100.

Also, I fixed the ordering of messages in general.
